### ISSUE TITLE
Fig not working despite all fig doctor checks passing

### ISSUE THREAD

**Issue by Tom-Ainsworth** [withfig/fig#2885]
### Checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my issue
- [X] I have run `fig doctor` in the affected terminal session
- [X] I have run `fig restart` and replicated the issue again

### Operating system

macOS 14.2.1 (23C71)

### Expected behaviour

Auto complete/prompts should appear

### Actual behaviour

No autocomplete/prompts, and an extra cursor appears either above or below my current line, depending on the terminal (below in vs code, above in macOS terminal)

### Steps to reproduce

I've run fig doctor and fig restart
uninstalled fig from the fig dashboard
reinstalled fig
setup shell integrations
reran fig doctor/restart
closed the terminal window and opened a new one, then reran the commands

still no luck

### Environment

```yaml
fig-details:
  - 2.17.0
hardware-info:
  - model: 
  - model-id: 
  - chip-id: Apple M1
  - cores: 8
  - mem: 16.00 GB
os-info:
  - macOS 14.2.1 (23C71)
environment:
  - shell: /bin/zsh
  - terminal: vscode
  - cwd: /Users/tomainsworth/Code/otsuka-hcp-portal
  - exe-path: /Users/tomainsworth/.fig/bin/fig
  - install-method: unknown
  - env-vars:
    - FIGTERM_SESSION_ID: 16f3f357-ea0a-4a2f-acbd-62499de4d214
    - FIG_PID: 18054
    - FIG_SET_PARENT: 16f3f357-ea0a-4a2f-acbd-62499de4d214
    - FIG_SET_PARENT_CHECK: 1
    - FIG_TERM: 2.17.0
    - PATH: /Users/tomainsworth/.nvm/versions/node/v18.16.0/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:/Users/tomainsworth/.nvm/versions/node/v18.16.0/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/tomainsworth/.fig/bin:/Users/tomainsworth/.local/bin:/Users/tomainsworth/.composer/vendor/bin:/Users/tomainsworth/.composer/vendor/bin
    - SHELL: /bin/zsh
    - TERM: xterm-256color
    - ZDOTDIR: /Users/tomainsworth
    - __CFBundleIdentifier: com.microsoft.VSCode
```

**Comment by shifenis** [comment]
I have same problem, switch from beta to stable but same problem. 
`fig doctor` passes all stages with success

**Comment by Tom-Ainsworth** [comment]
Sidenote: I've had Fig since May 2023, and got upgraded to the Pro tier when Amazon took over. It has been working fine for a long time, but today I've had a number of issues, this being the latest one

**Comment by ismael01866** [comment]
Yep, same issue. Started today,  prompt is not displaying. Ran fig doctor/restart (all checks passes, ...)

**Comment by patrikbily** [comment]
Same here.

**Comment by laurapadilla** [comment]
Same issue as well

**Comment by laurapadilla** [comment]
Same issue as well. Started today. Ran fig doctor (passing), fig restart and closed the terminal, still not working.

**Comment by navid-asgharzadeh** [comment]
same here

**Comment by laurapadilla** [comment]
> > same here
> > I also updated my Macbook to Sonoma 14.3 and then this happend. Don't know if it matters
> 
> Same here, updated today to Sonoma 14.3 (23D56) and stop working since that

I updated to Sonoma a few months ago (currently on `14.2.1 (23C71)`) and Fig was working just fine!

**Comment by elizatlawy** [comment]
Same here

**Comment by laurapadilla** [comment]
> > same here
> > I also updated my Macbook to Sonoma 14.3 and then this happend. Don't know if it matters
> 
> Same here, updated today to Sonoma 14.3 (23D56) and stop working since that

I'm still on 14.2.1 and everything was working fine until today. I'm thinking it's a Fig-specific issue?

**Comment by commonPlaceC** [comment]
same here, that's what web inspector says
![image](https://github.com/withfig/fig/assets/83515826/c6c6a29a-d7c5-46d2-9053-1e92c8581be9)

**Comment by GasserKhaled-Incorta** [comment]
Same here!

**Comment by SamJBarney** [comment]
This is also occurring on MacOS Ventura 13.6.3.

**Comment by mkirlin** [comment]
Checking in to note another instance of this on Sonoma 14.2.1 (23C71), fig_cli 2.17.0. `fig doctor` is reporting clean, but no autocomplete anywhere.

**Comment by themashcodee** [comment]
same here, all of a sudden it stopped working, everything is fine.

I tried reinstalling fig, restarting fig, restarting my laptop. But the suggestions does not comes up.

**Comment by nabetse28** [comment]
Same here, with MacOS Sonoma 14.3 intel chip

**Comment by 2coo** [comment]
Dropdown not appearing
https://github.com/withfig/fig/assets/40331144/e2da84f2-1a72-4295-8b99-5d6b04a6a43c

**Comment by 2coo** [comment]
Guys, they rebranded `Fig` as `CodeWhisperer`.

Solution: uninstall `Fig` but install `AWS CodeWhisperer`

**Comment by themashcodee** [comment]
@2coo I did that but CodeWhisperer does not support https://cursor.sh/

**Comment by 80imike** [comment]
Same here, with MacOs Sonoma 14.2.1

**Comment by tungamiraimujuru** [comment]
Also running MacOs Sonoma 14.2.1 and facing the same issue, since yesterday.

**Comment by akifunal** [comment]
I have been experiencing the same issue with macOS Sonoma 14.2.1 since yesterday.

**Comment by knielsen-foreflight** [comment]
I think they forgot to do some permission handling in the normal Fig. 
Installing CodeWhisperer instead which works.

**Comment by rostik32** [comment]
I have been using fig for a year and I encountered the same problem a few days ago.

I have not updated Mac OS in the past few days, I have version 14.1.2 (23B92) MacbookAir m1

**Comment by jumpogpo** [comment]
Same here, with MacOs Sonoma 14.3

**Comment by yaser-alazm** [comment]
Same here, Fig stopped working since yesterday on Mac.

**Comment by Tom-Ainsworth** [comment]
@knielsen-foreflight Thanks for letting me know about CodeWhisperer. I've also installed that and it's working as expected. It would still be good to get a Fig fix though. I imagine it's become a secondary concern now AWS have taken over and made their own version

**Comment by marklanham** [comment]
> Guys, they rebranded `Fig` as `CodeWhisperer`.
> 
> Solution: uninstall `Fig` but install `AWS CodeWhisperer`
> 
> https://aws.amazon.com/blogs/devops/introducing-amazon-codewhisperer-for-command-line/

Yep, I uninstalled Fig and installed CodeWhisperer and now it works again.

**Comment by yaser-alazm** [comment]
I can confirm too that it's working after installing CodeWhisperer

**Comment by laurapadilla** [comment]
Can confirm that it works after installing CodeWhisperer (on MacOS Sonoma 14.3 [23D56]). 

They do have a tiny little message on the [fig.io ](https://fig.io/) homepage announcing the change.

<img width="1398" alt="Screenshot 2024-01-26 at 9 23 03 AM" src="https://github.com/withfig/fig/assets/19558580/50dcde57-1eca-4819-859e-9534ffa1061b">

**Comment by danielweck** [comment]
Yep, same experience here, I uninstalled Fig and migrated to AWS CodeWhisperer.

**Comment by LawrenceZ** [comment]
did anyone get the autocomplete with history to work in the aws codewhisperer cli?

**Comment by shifenis** [comment]
With `CodeWhisperer` we can't use custom scripts

**Comment by favna** [comment]
Same issue here, sidegrading to codewhisper works. Shame for the no custom scripts though as I used one for work. I hope they add that back.

**Comment by themashcodee** [comment]
It started working now, I don't know how 🧐, maybe they fixed their servers or something.
But I am happy now :)

**Comment by HeathStratosphere** [comment]
> @2coo I did that but CodeWhisperer does not support https://cursor.sh/

Working in Cursor IDE for me:

```

Version: 0.24.3
VSCode Version: 1.85.1
Commit: 0924ede25947ac9b7d2e64a607a8c74cf8668f70
Date: 2024-01-28T07:15:14.673Z
Electron: 25.9.7
ElectronBuildId: undefined
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 22.5.0

```

**Comment by navid-asgharzadeh** [comment]
I just ran 'fig restart' and now it works again 🥳

**Comment by mschrage** [comment]
Hi everyone
**Quick Fix**
1. Run `fig restart`
2. If still not fixed, run `fig debug devtools autocomplete` and once the page is open hit `cmd+r` to refresh the page

**More information**
It appears that this is a caching issue related to macOS Sonoma. We aren't 100% sure and we are investigating. We are currently not able to reproduce it on our end. We are working with users to find out more

**Other**
If you're just using Fig for autocomplete, we recommend upgrading to [CodeWhisperer for command line](https://docs.aws.amazon.com/codewhisperer/latest/userguide/command-line.html)

**Comment by micheletedeschi** [comment]
I fixed changing zsh theme, using awesomepanda theme the autocomplete doesn't work

**Comment by dikaio** [comment]
9 months since this repository has been worked on. Like most other open-source projects bought out by FANGs, they end up dead or push users into one of their paid products/services.

**Comment by TvojTatk0** [comment]
I had very same issues with fig. Today, update showed up. Since update, even dashboard stop working with error message: "Something went wrong :(". 
I tried:
- reinstall
- a lot of restarts
- log in/out
- force quit

Do I really have to switch to the CodeWhisperer?
![image](https://github.com/withfig/fig/assets/28844291/4ff420b6-97af-40fb-9d9c-6a83016cffc2)

**Comment by beatrizsmerino** [comment]
I have the same problem 

<img width="287" alt="Captura de pantalla 2024-01-31 a las 12 59 34" src="https://github.com/withfig/fig/assets/14045148/febce54e-3dfb-45a2-9f23-902ba5564782">

the login does not work now, but a few days ago it was good

**Comment by blockiyt** [comment]
+1

**Comment by ben-girardet** [comment]
I tried AWS CodeWhisperer but it doesn't cover half of the features available in Fig (yet I presume). It's not a valid workaround for now.

**Comment by kachkaev** [comment]
Same here. Autocompletion works, but Dashboard does not load. Following https://github.com/withfig/fig/issues/2885#issuecomment-1915357652 does not help. Here is what I see in `fig debug devtools autocomplete`:

![Screenshot 2024-02-02 at 16 50 12](https://github.com/withfig/fig/assets/608862/b6f3d227-f5cf-4dea-820c-fe019f6a8aff)

**Comment by mschrage** [comment]
Thanks to everyone for chiming in! This thread has quite a few different issues being reported, and we think most have been resolved.

Please open a new issue if you are still being impacted.
